presto-native-execution/presto_cpp/main/QueryContextManager.cpp

```diff
@@ -156,7 +156,7 @@ void updateVeloxConnectorConfigs(
     // Do not retain cache if `node_selection_strategy` is explicitly set to
     // `NO_PREFERENCE`.
     auto it = connectorConfig.find("node_selection_strategy");
-    if (it == connectorConfig.end() || it->second == "NO_PREFERENCE") {
+    if (it != connectorConfig.end() && it->second == "NO_PREFERENCE") {
```

Comment:
So we assume the default is to use cache? Thanks!

Reply:
Yes. To unblock Deltoid